api/grpcserver/transaction_service.go

```diff
@@ -272,3 +280,135 @@ func (s TransactionService) TransactionsStateStream(in *pb.TransactionsStateStre
 		// that the service needs to shut down?
 	}
 }
+
+// StreamResults allows to query historical results and subscribe to live data using the same filter.
+func (s TransactionService) StreamResults(in *pb.TransactionResultsRequest, stream pb.TransactionService_StreamResultsServer) error {
+	var (
+		filter    transactions.ResultsFilter
+		sub       *events.BufferedSubscription[types.TransactionWithResult]
+		err       error
+		persisted types.LayerID
+	)
+	if len(in.Address) > 0 {
+		addr := types.BytesToAddress(in.Address)
+		filter.Address = &addr
+	}
+	if len(in.Id) > 0 {
+		var id types.TransactionID
+		copy(id[:], in.Id)
+		filter.TID = &id
+	}
+	if in.Start > 0 {
+		lid := types.NewLayerID(in.Start)
+		filter.Start = &lid
+	}
+	if in.End > 0 {
+		if in.Watch {
+			return status.Error(codes.InvalidArgument, "watch stream should have an empty End argument")
+		}
+		lid := types.NewLayerID(in.End)
+		filter.End = &lid
+	}
```

Comment:
what happens if none of the criteria is set?

Reply:
it constructs a query without `where` statement